modules/standard/IO.chpl

```diff
@@ -1971,7 +1961,54 @@ proc =(ref ret:channel, x:channel) {
 }
 
 pragma "no doc"
-proc channel.channel(param writing:bool, param kind:iokind, param locking:bool, f:file, out error:syserr, hints:c_int, start:int(64), end:int(64), in local_style:iostyle) {
+proc channel.init(param writing:bool, param kind:iokind, param locking:bool) {
```

Comment:
I feel a little surprised that this is the way you write the default initializer for a generic record. Is that something we've talked about in a GitHub issue somewhere? I probably just forgot everything...

Reply:
I thought private couldn't be applied to methods yet, am I misremembering?

Reply:
[See here](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/chips/10.rst#type-declarations-of-generic-instantiations), which was a result of discussion from back in May

Reply:
ah, thanks. I'm surprised that document doesn't describe default initialization at all - I was looking for a place discussion of default initialization could refer to the section you pointed to.

Reply:
Hmm, yeah, there's probably room for a crosslink in there...